From d628149fbd2606933c37a693040a7a17550cf44b Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Wed, 27 Aug 2025 16:56:40 +0200
Subject: [PATCH 042/553] util/unicode: createNormalizationCorrections():
 remove replace() call

Some users of the split()ed value handled intervening whitespace
already:

- fields[0..1] are piped through parseHex(), which does

- fields[2] is unused

- fields[3] needs to be trimmed, so do that.

As a consequence, we can accept the line by reference to const
QByteArray now.

Amends the start of the public history.

Pick-to: 6.9 6.8 6.5
Change-Id: I60371820cd143b980c81a1077d9c3e34528f1830
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit b98aa3983ce896d00c63ae06fa8b78a65fd8cec2)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 util/unicode/main.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/util/unicode/main.cpp b/util/unicode/main.cpp
index 73c39e6554e..aaa6e2f33ba 100644
--- a/util/unicode/main.cpp
+++ b/util/unicode/main.cpp
@@ -1745,9 +1745,7 @@ static QByteArray createNormalizationCorrections()
     int maxVersion = 0;
     int numCorrections = 0;
     readUnicodeFile("NormalizationCorrections.txt",
-                    [&] (QByteArray &line, int lineNo) {
-        line.replace(" ", "");
-
+                    [&] (const QByteArray &line, int lineNo) {
         Q_ASSERT(!line.contains(".."));
 
         QList<QByteArray> fields = line.split(';');
@@ -1756,6 +1754,7 @@ static QByteArray createNormalizationCorrections()
         NormalizationCorrection c = { 0, 0, 0 };
         c.codepoint = parseHex(fields[0], lineNo);
         c.mapped = parseHex(fields[1], lineNo);
+        fields[3] = std::move(fields[3]).trimmed();
         if (fields.at(3) == "3.2.0")
             c.version = QChar::Unicode_3_2;
         else if (fields.at(3) == "4.0.0")
-- 
2.51.2

